subworkflows/local/additional_annotation.nf

```diff
@@ -45,23 +45,36 @@ workflow ADDITIONAL_ANNOTATION {
     ch_snpeff_config = Channel.empty()
 
     SNPEFF_BUILD (
-        fasta,
+        fasta_path,
         ch_annot
     )
     ch_snpeff_db     = SNPEFF_BUILD.out.db
     ch_snpeff_config = SNPEFF_BUILD.out.config
     ch_versions      = ch_versions.mix(SNPEFF_BUILD.out.versions)
 
-    SNPEFF_ANN (
+    genome_id = fasta_path.map { input ->
+        def file = input instanceof List ? input.flatten()[0] : input
+        def filename = file.getName()
+        return filename.replaceAll(/\.f(ast|na)?(\.gz)?$/, '')
+    }
+
+    genome_ids = vcf.map { genome_id.value }
+
+    snpeff_db_value     = ch_snpeff_db.first()
+    snpeff_config_value = ch_snpeff_config.first()
+
+    snpeff_cache_per_sample = vcf.map { [ [ id: genome_id.value ], snpeff_db_value.value ] }
```

Comment:
Consider doing a `.combine` and a `.multiMap` as I find it challenging to follow. (especially with `genome_id` and `genome_idS`)

Also, in the previous module `fasta.basename` was used, is there a reason to make this more complex? 
I would suggest to do something like this:
```
SNPEFF_SNPEFF(
 vcf,
 fasta.map{f -> f.baseName}, 
 ch_snpeff_db.map{db -> [[:], db]}
 ch_snpeff_config
)
```
If  `ch_snpeff_db.map{db -> [[:], db]}` gives issues because it's not a value channel, then we could also modify the SNPEFF_SNPEFF as well to not take in meta2